src/common/engine.js

```diff
@@ -268,7 +268,7 @@ export class CurrencyEngine {
         )
       } catch (e) {
         this.log.error(
-          'Error writing to localDataStore. Engine not started:' + err
+          'Error writing to localDataStore. Engine not started:' + sāf(e)
```

Comment:
Instead of wrapping the error function with the "sāf" call every single time, in the constructor, instead of line 82, you should do something like this:

```js
this.log = {
   error: (message, e) => opts.log.error(message + sāf(e))
   warn: ...
   ....
 }
 ```